### ISSUE TITLE
Support ValueOption in FsToolkit.Errorhandling.asyncOption

### ISSUE THREAD

**Issue by Xyncgas** [demystifyfp/FsToolkit.ErrorHandling#225]
currently we can use ```let! x = ``` with say ``` int option``` but not ```int voption```
it seems voptions can reduce memory allocation with option types

**Comment by TheAngryByrd** [comment]
Yep I agree this is a good idea. Would you like to submit a pull request for it?

**Comment by Xyncgas** [comment]
Why do I have to do everything

**Comment by Xyncgas** [comment]
Maybe instead of making it possible by doing what OptionCE did, I can make asyncOption CE to use voption instead of converting voption to option

**Comment by Xyncgas** [comment]
in the future
